include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -540,6 +540,26 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
     AssertThrow(turbulence_model_constant > 0,
                 dealii::ExcMessage("parameter must be greater than zero"));
   }
+
+  // SIMPLEX ELEMENTS
+  if(grid.element_type == ElementType::Simplex)
+  {
+    AssertThrow(temporal_discretization == TemporalDiscretization::BDFCoupledSolution,
+                dealii::ExcMessage("Only BDFCoupledSolution is supported for simplex elements."));
+
+    AssertThrow(preconditioner_pressure_block == SchurComplementPreconditioner::None or
+                  preconditioner_pressure_block == SchurComplementPreconditioner::LaplaceOperator,
+                dealii::ExcMessage(enum_to_string(preconditioner_pressure_block) +
+                                   " is not supported with simplex elements."));
+
+    AssertThrow(not(preconditioner_velocity_block == MomentumPreconditioner::InverseMassMatrix),
+                dealii::ExcMessage(enum_to_string(preconditioner_velocity_block) +
+                                   " is not supported for simplex elements."));
+
+    AssertThrow(not(preconditioner_projection == PreconditionerProjection::InverseMassMatrix),
+                dealii::ExcMessage(enum_to_string(preconditioner_projection) +
+                                   " is not supported for simplex elements."));
+  }
```

Comment:
I have asserted everything, which don't work with simplices, here.  I couldn't decide if I should go for a design similar to `poisson/application.h `, where we have a simplex configuration and a hypercube configuration.

Reply:
I think the asserts are also needed where the actual implementations happen. How can a developer be safe to remove such an assert in `Parameters` if this is the only assert on a certain topic/functionality?